c2corg_ui/static/js/card.js

```diff
@@ -164,13 +244,64 @@ app.CardController.prototype.createURL = function() {
   var loc = window.location.pathname;
   // Don't create links on edit and add pages.
   if (loc.indexOf('/edit/') === -1 && loc.indexOf('/add') === -1) {
-    return this.url_.buildDocumentUrl(
-      this.type, this.doc['document_id'], this.doc['locales'][0]);
+    if (this.type == 'feeds') {
+      return this.url_.buildDocumentUrl(
+        app.utils.getDoctype(this.doc['document']['type']), this.doc['document']['document_id'], this.doc['document']['locales'][0]);
+    } else {
+      return this.url_.buildDocumentUrl(
+        this.type, this.doc['document_id'], this.doc['locales'][0]);
+    }
```

Comment:
```
var type, doc;
if (this.type == 'feeds') {
  type = app.utils.getDoctype(this.doc['document']['type']);
  doc = this.doc['document'];
} else {
  type = this.type;
  doc = this.doc;
}
return this.url_.buildDocumentUrl(type, doc['document_id'], doc['locales'][0]);
```

By the way isn't it a bit confused to have a ``document`` property in a ``document`` object (``document['document']``)?